openapi-v1.yaml

```diff
@@ -6041,3 +6148,180 @@ paths:
+      responses:
+        204:
+          description: group created successfully
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+
+  /groups/{namespace}/{name}:
+    parameters:
+      - name: namespace
+        type: string
+        in: path
+        required: true
+        description: The namespace of the group
+      - name: name
+        type: string
+        in: path
+        required: true
+        description: The name of the group
+    post:
+      description: Adds a member to a group
+      operationId: addMember
+      tags:
+        - groups
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: '#/definitions/GroupMember'
+      responses:
+        204:
+          description: member added successfully
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+    get:
+      description: Returns the contents, assets and subgroups, of the group
+      operationId: listGroup
+      tags:
+        - groups
+      parameters:
+        - name: output
+          in: query
+          required: false
+          type: string
+          enum:
+            - attributes # return only the attributes of the group
+            - groups     # return only the attributes and one page of subgroups
+            - assets     # return the attributes and one page of subgroups and assets
+        - name: page
+          in: query
+          description: pagination offset for assets
+          type: integer
+          required: false
+        - name: per_page
+          in: query
+          description: pagination limit for assets
+          type: integer
+          required: false
+      responses:
+        200:
+          description: the group contents
+          schema:
+            $ref: '#/definitions/GroupListing'
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+    patch:
+      description: Changes attributes of the group
+      operationId: updateGroup
+      tags:
+        - groups
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: '#/definitions/GroupUpdate'
+      responses:
+        204:
+          description: attributes changed successfully
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+    delete:
+      description: Deletes the group and all the subgroups recursively. The assets are not deleted and are not relocated to any other group
+      operationId: deleteGroup
+      tags:
+        - groups
+      responses:
+        204:
+          description: group deleted successfully
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+
+  /groups/{namespace}/{name}/{member_namespace}/{member_name}:
```

Comment:
If we adopt asset instead of member as mentioned, maybe we can make it `/groups/{namespace}/{name}/{member_namespace}/{member_name}`?

Reply:
Probably you mean `asset_namespace` and `asset_name` :+1:

Reply:
right